server/src/main/java/org/elasticsearch/search/aggregations/metrics/ValueCountAggregator.java

```diff
@@ -125,6 +126,11 @@ public InternalAggregation buildAggregation(long bucket) {
         return new InternalValueCount(name, counts.get(bucket), metadata());
     }
 
+    @Override
+    public ScoreMode scoreMode() {
+        return valuesSource != null && valuesSource.needsScores() ? ScoreMode.COMPLETE : ScoreMode.COMPLETE_NO_SCORES;
+    }
+
```

Comment:
Note that this construct seems to be used by all `NumericMetricsAggregator`, I winder if we can move it to the parent class?

Reply:
Probably!